EventFilter/CTPPSRawToDigi/interface/TotemSampicFrame.h

```diff
@@ -0,0 +1,263 @@
+/****************************************************************************
+*
+* This is a part of the TOTEM offline software.
+* Authors:
+*   Nicola Minafra
+*
+****************************************************************************/
+
+#ifndef EventFilter_CTPPSRawToDigi_TotemSampicFrame
+#define EventFilter_CTPPSRawToDigi_TotemSampicFrame
+
+#include <vector>
+#include <cstddef>
+#include <cstdint>
+#include <iomanip>
+#include <bitset>
+
+#include "EventFilter/CTPPSRawToDigi/interface/VFATFrame.h" 
+
+enum TotemSampicConstant {
+  NumberOfSamples = 24,
+};
+
+#pragma pack(push,1)
```

Comment:
@Dr15Jones 
are these pragmas well supported?

Reply:
I have no idea. I’ve never seen them before. I believe there are standard C++ ways of doing such things now.

Reply:
Thanks for the comment! I was indeed wondering myself. To my knowledge, it is supported by gcc since many years now; there might be some problem using pointers to "non aligned" objects, which we don't do. 
I used this method since it is efficient (no copies) and it is extremely practical to have the dataformat encoded in one place only: we changed the position of the information several times while developing the firmware and i was able to adapt the code quickly and safely only changing the struct. Now the firmware format should be definitive. 
That said, if you have suggestions I'll be happy to improve the code!